src/transferslot.cpp

```diff
@@ -310,21 +322,31 @@ void TransferSlot::doio(MegaClient* client)
 
                             return transfer->failed(API_EINTERNAL);
                         }
+
+                        errorcount = 0;
+                        transfer->failcount = 0;
```

Comment:
I agree, but it's the current behavior in `master` and I think that it's better to not change it during this pull request, since it's a different issue that should be analyzed and tested.